<launch>

	<node pkg="semantic_mesh_loam" type="classify_pointcloud" name="classify_pointcloud" output="screen">
		<param name="lidar" value="HDL-64E" />
	</node>

	<node pkg="semantic_mesh_loam" type="laser_odometry" name="laser_odometry" output="screen">
		<param name="systemdelay" value="50" type="int"/>
		<param name="MaxCorrespondDistance" value="0.40" type="double"/>
	</node>

	<arg name="PCDpath" default="/home/amsl/PCD_data/sequence00_velodyne/" />
	<arg name="PCDfilename" default="semantic_mesh_loam" />

	<node pkg="semantic_mesh_loam" type="laser_mapping" name="laser_mapping" output="screen">
        
        <param name="ReferenceFrame" value="vehicle" />

        <param name="scancounter" value="30" type="int"/>

		<param name="systemdelay" value="5" type="int" />
		<param name="filepath" value="$(arg PCDpath)" />
		<param name="filename" value="$(arg PCDfilename)" />

		<param name="PCDsaveChecker" value="true" type="bool" />
		<param name="ROSpublishChecker" value="true" type="bool" />

		<param name="PCDsaveAscii" value="true" type="bool" />
		<param name="PCDsaveBinary" value="false" type="bool" />

	</node>

	<arg name="path" default="/home/amsl/bag_data/" />
	<arg name="file_1" default="semantickitti_sequence00" />
	<arg name="play_speed" default="-r 0.2" />

	<node pkg="rosbag" type="play" name="rosbag_play" output="screen"
	args="$(arg path)$(arg file_1).bag $(arg play_speed)" />

	<node pkg="rviz" type="rviz" name="rviz" args="-d $(find semantic_mesh_loam)/config/test_laser_mapping.rviz"/>

</launch>

